helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -444,6 +456,13 @@ private void refreshBaseline(ResourceControllerDataProvider clusterData,
         optimalAssignment.getOptimalResourceAssignment();
 
     LOG.info("Finish calculating an assignment. Took: {} ms.", System.currentTimeMillis() - startTime);
+
+    BaselineDivergenceGauge baselineDivergenceGauge = _metricCollector.getMetric(
```

Comment:
Can you please move this to partialRebalance()? This is not the right place to check.